File: Sort/quicksort.cpp
#include<cstdio>
#include<vector>
using namespace std;
vector<int> v;
//快速排序
int division(vector<int> &data, int low, int high) {
  int temp = data[low];
  while(low < high) {
    while(low < high && data[high] >= temp) high--;
    data[low] = data[high];
    while(low < high && data[low] <= temp) low++;
    data[high] = data[low];
    data[low] = temp;
  }
  return low;
}
void quicksort(vector<int> &data, int low, int high) {
  if(low < high) {
    int i = division(data, low, high);
    quicksort(data, low, i - 1);
    quicksort(data, i + 1, high);
  }
}
int main()
{
  int n;
  scanf("%d", &n);
  v.resize(n);
  for(int i = 0; i < n; i++)
    scanf("%d", &v[i]);
  quicksort(v, 0, v.size() - 1);
  for(int i = 0; i < n; i++) {
    if(i == 0)
      printf("%d", v[i]);
    else
      printf(" %d", v[i]);
  }
  return 0;
}

File: Sort/quicksort.py
def partition(arry, low, high):
    i = low - 1
    pivot = arry[high]
    for j in range(low , high):
        if arry[j] <= pivot:
            i = i + 1
            arry[i], arry[j] = arry[j], arry[i]
    arry[i + 1], arry[high] = arry[high], arry[i + 1]
    return (i + 1)

def QuickSort(arry, low, high):
    if low < high:
        pi = partition(arry, low, high)
        QuickSort(arry, low, pi - 1)
        QuickSort(arry, pi + 1, high)

arr = [10, 3, 18, 9, 4, 7, 15]
n = len(arr)
QuickSort(arr, 0, n - 1)
print ("The Sorted array is: ")
for i in range(n):
    print ("%d" %arr[i]),
